hamster_gtk/preferences/widgets/config_combo_box_text.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of 'hamster-gtk'.
+#
+# 'hamster-gtk' is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# 'hamster-gtk' is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with 'hamster-gtk'.  If not, see <http://www.gnu.org/licenses/>.
+
+"""This module provides ComboBoxText widget extended to store preferences."""
+
+from __future__ import absolute_import
+
+from gi.repository import Gtk
+
+from .config_widget import ConfigWidget
+
+
+class ConfigComboBoxText(Gtk.ComboBoxText, ConfigWidget):
+    """A combo box that stores configuration."""
+
+    # Required else you would need to specify the full module
+    # name in ui file
+    __gtype_name__ = 'ConfigComboBoxText'
+
+    def __init__(self, values=None):
+        """Initialize widget."""
+        super(Gtk.ComboBoxText, self).__init__()
+
+        if values is not None:
+            for id, value in values:
```

Comment:
If it is true that ``id`` can be an arbitrary string, maybe a better variable name can be chosen. Whilst ID's of cause do not need to be numeric, its a common first thought I think.